sklearn/preprocessing/imputation.py

```diff
@@ -299,13 +312,69 @@ def _dense_fit(self, X, strategy, missing_values, axis):
 
             return most_frequent
 
+    def _sparse_transform(self, X, missing_values, axis, valid_stats, valid_idx):
+        """transformer on sparse data."""
+        mask = _get_mask(X.data, missing_values)
+        indexes = np.repeat(np.arange(len(X.indptr) - 1, dtype=np.int),
+                            np.diff(X.indptr))[mask]
+
+        X.data[mask] = astype(valid_stats[indexes], X.dtype,
+                              copy=False)
+
+        if self.add_missing_indicator:
+            mask_matrix = X.__class__((mask, X.indices.copy(),
+                                      X.indptr.copy()), shape=X.shape,
+                                      dtype=X.dtype)
+            mask_matrix.eliminate_zeros()  # removes explicit False entries
+            features_with_missing_values = np.where(sparse_min_max
```

Comment:
True. Indeed `mask_matrix.sum(axis=0).nonzero()[1]` would do. [Indeed if it's a CSC matrix we can do `np.diff(mask_matrix.indptr)` and if CSR we can do `np.bincount(mask_matrix.indices, minlength=mask_matrix.shape[1])`, as implemented in `mask_matrix.getnnz(axis=0)` in recent scipy. So many ways to skin this cat, and for some reason I thought of max and got hooked on it.]